runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -298,30 +298,42 @@ public UUID getStreamID() {
     }
 
     private <R> R TXExecuteInner(Supplier<R> txFunction, boolean isMetricsEnabled) {
-        long sleepTime = 1L;
         final long maxSleepTime = 1000L;
+        long sleepTime = 1L;
         int retries = 1;
+
         while (true) {
             try {
                 rt.getObjectsView().TXBegin();
                 R ret = txFunction.get();
                 rt.getObjectsView().TXEnd();
                 return ret;
             } catch (TransactionAbortedException e) {
-                // If TransactionAbortedException is due to a 'Network Exception' do not keep retrying a nested
-                // transaction indefinitely (this could go on forever). If this is part of an outer transaction abort
-                // and remove from context. Re-throw exception to client.
-                if (e.getAbortCause() == AbortCause.NETWORK) {
-                    if (TransactionalContext.getCurrentContext() != null) {
+                // If TransactionAbortedException is due to a 'Network Exception' or a
+                // 'Unsupported Operation Exception', do not keep retrying a nested transaction
+                // indefinitely (this could go on forever). If this is part of an outer
+                // transaction abort and remove from context. Re-throw exception to client.
+                if (TransactionalContext.getCurrentContext() != null) {
+                    if (e.getAbortCause() == AbortCause.NETWORK) {
                         TransactionalContext.getCurrentContext().abortTransaction(e);
                         TransactionalContext.removeContext();
                         throw e;
                     }
+
+                    if (e.getAbortCause() == AbortCause.UNSUPPORTED_OPERATION) {
+                        // No need to deal with TransactionalContext in this case.
+                        throw e;
+                    }
                 }
 
                 if (retries == 1) {
```

Comment:
I know you didnt write this code, but this will result in incorrect stats, since your fixing retries maybe also address that too?